packages/build/tests/tracing/tests.js

```diff
@@ -11,6 +17,70 @@ test('Tracing set multi span attributes', async (t) => {
   t.is(baggage.getEntry('foo').value, 'bar')
 })
 
+const testMatrixBaggageFile = [
+  {
+    description: 'when baggageFilePath is blank',
+    input: {
+      baggageFilePath: '',
+      baggageFileContent: null,
+    },
+    expects: {
+      somefield: undefined,
+      foo: undefined,
+    },
+  },
+  {
+    description: 'when baggageFilePath is set but file is empty',
+    input: {
+      baggageFilePath: BAGGAGE_PATH,
+      baggageFileContent: '',
+    },
+    expects: {
+      somefield: undefined,
+      foo: undefined,
+    },
+  },
+  {
+    description: 'when baggageFilePath is set and has content',
+    input: {
+      baggageFilePath: BAGGAGE_PATH,
+      baggageFileContent: 'somefield=value,foo=bar',
+    },
+    expects: {
+      somefield: { value: 'value' },
+      foo: { value: 'bar' },
+    },
+  },
+]
+
+testMatrixBaggageFile.forEach((testCase) => {
+  test.serial(`Tracing baggage loading - ${testCase.description}`, async (t) => {
+    const { input, expects } = testCase
+    if (input.baggageFilePath.length > 0) {
+      await mkdir(FIXTURES_DIR, { recursive: true })
+      await writeFile(input.baggageFilePath, input.baggageFileContent)
+    }
+
+    const ctx = loadBaggageFromFile(input.baggageFilePath)
+    const baggage = propagation.getBaggage(ctx)
+
+    Object.entries(expects).forEach(([property, expected]) => {
+      if (input.baggageFilePath === '') {
+        t.is(baggage, undefined)
+      } else {
+        if (expected === undefined) {
+          t.is(baggage.getEntry(property), expected)
+        } else {
+          t.is(baggage.getEntry(property).value, expected.value)
+        }
+      }
+    })
+    if (input.baggageFilePath.length > 0) {
+      rm(input.baggageFilePath, { force: true })
+    }
```

Comment:
We might want to move this to `test.after` to make sure we run it even if the assertions fail.

Reply:
addressed in: https://github.com/netlify/build/pull/5343/commits/29c85648418afef5e9b22bc781c482e52e61ce03